### PR TITLE
interop-testing: support dynamic configuration and accumulated stats for xDS test client

### PR DESCRIPTION
**Pull request by voidzcy**

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java [hunk 1]
```diff
@@ -167,8 +169,10 @@ private void parseArgs(String[] args) {
               + c.qps
               + "\n  --rpc=STR              Types of RPCs to make, ',' separated string. RPCs can "
               + "be EmptyCall or UnaryCall. Default: UnaryCall"
+              + "\n[deprecated] Use XdsUpdateClientConfigureService"
```
Comment (dapengzhang0) [inline interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:172]: I don't understand this. How to use `XdsUpdateClientConfigureService` to change the value, when `runQps()` starts immediately?

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:172]: By default, the test client just starts up and sends unary RPCs. That's the warmup phase, the test case won't begin until the test driver has checked if there is at least one RPCs succeed on each backend. This is to ensure the backend service configuration has been picked up by TD. We do not care what type of RPCs used to do the warm up, default unary is completely fine. After the warmup phase, we can change call behaviors to whatever needed for achieving the test goal.

Comment (dapengzhang0) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:172]: But `rpcTypes` is only read once in `runQps()`, how can `rpcTypes` be used after updated by the driver?

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:172]: Oops, this is screwed up. Currently a `PeriodicRpc` task is created per RpcType when the test client starts. It should be changed to run a single `PeriodicRpc` and each time it runs, it sends out one RPC per RpcType.

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:174]: Should be fixed now.

File: interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java [hunk 2]
```diff
@@ -364,31 +375,43 @@ public void onNext(SimpleResponse response) {
...
-      ListenableScheduledFuture<?> future =
-          exec.scheduleAtFixedRate(
-              new PeriodicRpc(rpcType), 0, nanosPerQuery, TimeUnit.NANOSECONDS);
+      private void handleRpcCompleted(long requestId, RpcType rpcType, String hostname,
+          Set<XdsStatsWatcher> watchers) {
+        int count = 0;
+        if (rpcsSucceededByPeer.containsKey(hostname)) {
+          count = rpcsSucceededByPeer.get(hostname);
+        }
+        rpcsSucceededByPeer.put(hostname, count + 1);
```
Comment (ericgribkoff) [inline interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:388]: Did I miss some synchronization barrier here? It seems like multiple calls to handleRpcCompleted could be executing simultaneously and `count` might no longer be correct by the time this `#put()` is invoked.

**Review by ericgribkoff — COMMENTED** [review]
Just made a quick pass, but the `RpcConfig` type and how it's updated with the config service aligns with my understanding of the spec for the tests. Just had a comment/concern about synchronization for one of the new stats.

File: interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java (hunk 2 above)
Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:388]: You are right, this needs to be fixed. 

Btw, now that we are warming up the test client normally before making calls open for testing. So we can still use the existing way for checking if there is at least one RPCs succeeding on each backend (getting the stats for next `n` RPCs). Then we do not need to categorize the number of successful RPCs for each peer, we just need to know the total successful RPCs.

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java]: Updated to count the total number of RPCs succeeded without counting them separately for each remote peer.

**Comment by voidzcy** [comment]
@ericgribkoff Ping.

**Review by ericgribkoff — APPROVED** [review]

**Pull request merged**